### PR TITLE
Null check before trying to fix on save

### LINKED ISSUE #50: Uncaught TypeError: Path must be a string. Received null
**Atom Version**: 1.12.4
**Electron Version**: 1.3.9
**System**: Microsoft Windows 8.1
**Thrown From**: [linter-xo](https://github.com/sindresorhus/atom-linter-xo) package, v0.20.0

### Stack Trace

Uncaught TypeError: Path must be a string. Received null

```
At G:\Applications\Atom\app-1.12.4\resources\app.asar\src\pane.js:1030

TypeError: Path must be a string. Received null
    at assertPath (path.js:7:11)
    at Object.join (path.js:466:7)
    at findPkg (file:///G:/Applications/Atom/.atom/packages/linter-xo/index.js:30:26)
    at file:///G:/Applications/Atom/.atom/packages/linter-xo/index.js:161:19
    at Function.module.exports.Emitter.simpleDispatch (G:\Applications\Atom\app-1.12.4\resources\app.asar\node_modules\event-kit\lib\emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (G:\Applications\Atom\app-1.12.4\resources\app.asar\node_modules\event-kit\lib\emitter.js:129:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (G:\Applications\Atom\app-1.12.4\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:1132:20)
    at TextBuffer.module.exports.TextBuffer.save (G:\Applications\Atom\app-1.12.4\resources\app.asar\node_modules\text-buffer\lib\text-buffer.js:1124:19)
    at TextEditor.module.exports.TextEditor.save (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\text-editor.js:871:26)
    at Pane.module.exports.Pane.saveItem (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\pane.js:738:18)
    at Pane.saveItem (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\pane.js:3:59)
    at Pane.module.exports.Pane.saveActiveItem (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\pane.js:721:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\workspace.js:693:35)
    at atom-workspace.core:save (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\register-default-commands.js:222:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\command-registry.js:259:29)
    at G:\Applications\Atom\app-1.12.4\resources\app.asar\src\command-registry.js:3:59
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (G:\Applications\Atom\app-1.12.4\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:587:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (G:\Applications\Atom\app-1.12.4\resources\app.asar\node_modules\atom-keymap\lib\keymap-manager.js:382:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\window-event-handler.js:106:36)
    at HTMLDocument.<anonymous> (G:\Applications\Atom\app-1.12.4\resources\app.asar\src\window-event-handler.js:3:59)
```

### Commands

```
  2x -0:37.2.0 core:save (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "autoHideMenuBar": true,
    "automaticallyUpdate": false,
    "closeEmptyWindows": false,
    "disabledPackages": [
      "tab-preview",
      "file-icons"
    ],
    "openEmptyEditorOnStart": false,
    "telemetryConsent": "limited",
    "themes": [
      "flexible-ui",
      "bionis-dark-syntax"
    ],
    "warnOnLargeFileLimit": 2
  }
}
```

### Installed Packages

```coffee
# User
atom-hipster-ui, v0.1.0 (inactive)
atom-html-preview, v0.1.22 (active)
atom-live-server, v1.2.5 (inactive)
atom-material-syntax-dark, v0.2.5 (inactive)
atom-material-syntax-light, v0.4.5 (inactive)
atom-material-ui, v1.3.6 (inactive)
atom-minify, v0.7.5 (active)
atomic-ui, v1.5.2 (inactive)
bionis-dark-syntax, v1.1.0 (active)
bionis-syntax, v1.1.0 (inactive)
chrome-color-picker, v0.8.0 (inactive)
emmet, v2.4.3 (active)
ficus-dark-syntax, v1.1.0 (inactive)
ficus-syntax, v1.1.0 (inactive)
flexible-ui, v0.10.5 (active)
genesis-ui, v0.5.0 (inactive)
github-atom-dark-syntax, v0.2.0 (inactive)
github-atom-light-syntax, v0.2.0 (inactive)
isotope-light-ui, v2.2.6 (inactive)
isotope-ui, v2.7.1 (inactive)
linter, v1.11.18 (active)
linter-csslint, v1.3.3 (active)
linter-write-good, v0.8.0 (active)
linter-xo, v0.20.0 (active)
minimap, v4.25.6 (active)
minimap-cursorline, v0.2.0 (active)
minimap-find-and-replace, v4.5.1 (active)
nord-atom-ui, v0.8.0 (inactive)
nucleus-dark-ui, v0.10.15 (inactive)
peacock-syntax, v1.4.0 (inactive)
pigments, v0.37.0 (active)
smart-tab-name, v1.0.5 (active)
solarized-dark-ui, v0.3.1 (inactive)
solarized-light-ui, v0.1.0 (inactive)
steam-pirate-ui, v0.6.1 (inactive)
svg-preview, v0.10.0 (active)
tab-title, v0.3.3 (active)
unfancy-file-icons, v0.10.0 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.45.0 (inactive)
base16-tomorrow-dark-theme, v1.3.0 (inactive)
base16-tomorrow-light-theme, v1.3.0 (inactive)
one-dark-ui, v1.6.2 (inactive)
one-light-ui, v1.6.2 (inactive)
one-dark-syntax, v1.5.0 (inactive)
one-light-syntax, v1.5.0 (inactive)
solarized-dark-syntax, v1.0.5 (inactive)
solarized-light-syntax, v1.0.5 (inactive)
about, v1.7.0 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.13.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.4 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.2 (active)
command-palette, v0.39.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.202.2 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.158.8 (active)
metrics, v1.1.2 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.1 (inactive)
settings-view, v0.243.1 (active)
snippets, v1.0.4 (active)
spell-check, v0.68.4 (active)
status-bar, v1.4.1 (active)
styleguide, v0.47.2 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.210.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.0 (active)
language-csharp, v0.13.0 (active)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (active)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.46.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

### PR DESCRIPTION
**Pull request by AntonNiklasson**
Fixes #50.

### CODE REVIEW

**Comment by jonathandelgado** [comment]
Can confirm this resolves #50

**Comment by sindresorhus** [comment]
Thanks for fixing @AntonNiklasson :)

File: index.js [hunk 1]
```diff
@@ -160,9 +160,14 @@ export function activate() {
 			editor.getBuffer().onWillSave(() => {
 				const isJS = SUPPORTED_SCOPES.includes(editor.getGrammar().scopeName);
 				const shouldFixOnSave = atom.config.get('linter-xo.fixOnSave');
-
 				const filePath = editor.getPath();
-				const {pkg} = findPkg(pkgDir.sync(path.dirname(filePath)));
+				const dir = pkgDir.sync(path.dirname(filePath));
```
Comment (jfmengels) [inline index.js]: This sync operation doesn't have to be made if `!shouldFixOnSave || !isJS`. It would be better perf-wise to abort early.

Comment (AntonNiklasson) [inline index.js]: Dude, that's true. I could PR that if you won't.

Comment (jfmengels) [inline index.js]: That'd be great! Thanks!

**Pull request merged**